src/events/components/SearchModule/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { DateTime } from 'luxon';
+import React, { FC } from 'react';
+import ToggleSwitch from '../../../common/components/ToggleSwitch';
+import { useQueryParam } from '../../../common/hooks/useQueryParam';
+import {
+  defaultDateEndParam,
+  defaultDateStartParam,
+  defaultEventTypesParam,
+  defaultSearchParam,
+} from '../../../core/hooks/useQueryParamsState';
+import style from '../../../profile/components/Search/search.less';
+import DateInput from './DateInput';
+import { SelectMultiple } from './SelectMultiple';
+
+const SearchModule: FC = () => {
+  const [search, setSearch] = useQueryParam('search');
+  const [dateStart, setDateStart] = useQueryParam('dateStart');
+  const [dateEnd, setDateEnd] = useQueryParam('dateEnd');
+  const [eventTypes, setEventTypes] = useQueryParam('eventTypes');
+  const [attendanceEventsChecked, setAttendanceEventsChecked] = useQueryParam('attendanceEvents');
+
+  return (
+    <div className={style.grid}>
+      <input
+        className={style.searchInput}
+        type="search"
+        defaultValue={search || defaultSearchParam}
+        placeholder="Søk"
+        onChange={(event) => setSearch(event.target.value)}
+      />
+      <DateInput
+        label="Fra: "
+        time={DateTime.fromISO(dateStart || defaultDateStartParam).toFormat('yyyy-MM-dd')}
+        onChange={(event) => setDateStart(event.target.value)}
+      />
+      <DateInput
+        label="Til: "
+        time={DateTime.fromISO(dateEnd || defaultDateEndParam).toFormat('yyyy-MM-dd')}
+        onChange={(event) => setDateEnd(event.target.value)}
+      />
+      <SelectMultiple
+        onEventTypesInput={(event) =>
+          setEventTypes(
+            JSON.stringify(
+              [...event.target.options]
+                .filter((eventType) => eventType.selected)
+                .map((eventType) => parseInt(eventType.value, 10))
```

Comment:
Larger functions like this should probably not be inline in the JSX.
Try to keep as little logic as possible in the JSX, and rather as functions in the component body.